frontend/src/components/common/Admin/RemoveUserModal.tsx

```diff
@@ -1,33 +1,76 @@
+import { gql, useMutation, useQuery } from "@apollo/client";
 import {
   Button,
+  Divider,
   Modal,
   ModalContent,
   ModalFooter,
   ModalOverlay,
   useDisclosure,
 } from "@chakra-ui/react";
-import React from "react";
+import React, { useContext } from "react";
+import AuthContext from "../../../contexts/AuthContext";
 import { CloseOutlineIcon } from "../icons";
-import ModalText from "../ModalText";
+import ModalText from "./ModalText";
 import RemoveUserConfirmationModal from "./RemoveUserConfirmationModal";
+import RemoveUserErrorModal from "./RemoveUserErrorModal";
+import { Role } from "../../../types/AuthTypes";
 
 interface RemoveUserModalProps {
   name: string;
   email: string;
   onCloseParent: () => void;
 }
 
+interface UserGQLObject {
+  userByEmail: { id: string };
+}
+
+const GET_USER = gql`
+  query RemoveUserModal_UserByEmail($email: String!) {
+    userByEmail(email: $email) {
+      id
+    }
+  }
+`;
+
+const REMOVE_USER = gql`
+  mutation RemoveUserModal_DeleteUserByEmail($email: String!) {
+    deleteUserByEmail(email: $email)
+  }
+`;
+
 const RemoveUserModal = ({
   name,
   email,
   onCloseParent,
 }: RemoveUserModalProps): React.ReactElement => {
   const { onOpen, onClose, isOpen } = useDisclosure();
   const [showConfirmation, setShowConfirmation] = React.useState(false);
+  const [showError, setShowError] = React.useState(false);
+  const [removeUser, { error }] = useMutation<{ removeUser: null }>(
```

Comment:
do we need to include `refetchQuery` here? to refresh the list of admin users shown. more context [here](https://www.apollographql.com/docs/react/data/refetching/).